### PR TITLE
chore: make error message of cargo nightly fmt check more readable

### PR DESCRIPTION
**Pull request by yihau**

### CODE REVIEW

**Comment by yihau** [comment]
After reviewing, I will remove the XXX commit.

**Comment by yihau** [comment]
now it looks like this

https://buildkite.com/solana-labs/solana/builds/81421#01831b6d-bb6b-42b4-85f2-f658480c3ec0/2848-2870

File: ci/test-checks.sh [hunk 1]
```diff
@@ -86,7 +86,13 @@ nightly_clippy_allows=(
    "${nightly_clippy_allows[@]}"
 
 _ scripts/cargo-for-all-lock-files.sh -- nightly sort --workspace --check
-_ scripts/cargo-for-all-lock-files.sh -- nightly fmt --all -- --check
+_ scripts/cargo-for-all-lock-files.sh -- nightly fmt --all -- --check | tee nightly_fmt_check_result
```
Comment (t-nelson) [inline ci/test-checks.sh:89]: what's the tee for?

Comment (yihau) [reply ci/test-checks.sh:89]: we set -e in the begin. I would like to hide the error to avoid it exit immediately. Maybe there is another better way 🤔

Comment (yihau) [reply ci/test-checks.sh:89]: @t-nelson what do you think? If you don't oppose, I'm going to clean and merge this PR. or maybe we don't need this feature anymore? 🙊

**Pull request closed without merging**